crates/bevy_math/src/rects/urect.rs

```diff
@@ -309,8 +313,14 @@ impl URect {
     #[inline]
     pub fn inset(&self, inset: i32) -> Self {
         let mut r = Self {
-            min: (self.min.as_ivec2() - inset).as_uvec2(),
-            max: (self.max.as_ivec2() + inset).as_uvec2(),
+            min: UVec2::new(
+                self.min.x.checked_add_signed(-inset).unwrap(),
```

Comment:
Sorry I should have mentioned that earlier, but I'm not sure panicking here is the best thing because it's really not actionable. To avoid a panic, a user would have to write that exact same code again outside the method to check if the method would panic or not. I think all other methods like this just return an empty rectangle, which make sense I think (IIRC in CSS if your inner padding is larger than the content then simply there's 0 space anymore for content, but it doesn't crash the UI). So I think we'd be better with saturating operations here, that won't panic. The other methods that panic do so because of trivially wrong arguments like negative size